### ISSUE TITLE
Enhance Mashery Publishing Process

### ISSUE THREAD

**Issue by jpollock** [TIBCOSoftware/mashling#67]
During publish process:

1. allow for creation of io docs.
2. allow for creation of test package and plan and key
3. allow for setting api defaults in external file
4. generate sample curl calls
5. provide link to the control center page for the generated api
6. allows for updating api definition (re-publishing)

**Comment by jpollock** [comment]
example:

`mashling publish  -f /Users/jeremypollock/temp/mashling/rest-conditional-gateway.mashling.json -k <v3 key> -s <v3 secret> -u <username> -p <password> -areaId <areaid> -areaDomain tibcobanqio.api.mashery.com -mock false -h petstore.swagger.io -iodocs true -testplan true -apitemplate /Users/jeremypollock/temp/mashling/myapi_template.mashery.json`

**Comment by jpollock** [comment]
mashery template json might look like:
`{
    "api": {

        "endpoint": {
           "apiKeyValueLocationKey": "api_key",
           "requestAuthenticationType": "apiKeyAndSecret_SHA256",
           "outboundTransportProtocol": "https",
           "inboundSslRequired": true,
           "numberOfHttpRedirectsToFollow": 1

        },
        "qpsLimitOverall": 10
    },
    "package": {
        "sharedSecretLength": 10,
        "plan": {
            "selfServiceKeyProvisioningEnabled": false,
            "numKeysBeforeReview": 2
        }
    }
    
}`

**Comment by rkiesler1** [comment]
What is the template file used for? Does it contain any placeholders/replacement variables?

**Comment by jpollock** [comment]
Yes. Replacement values.
[myapi_template.mashery.json.zip](https://github.com/TIBCOSoftware/mashling/files/1451592/myapi_template.mashery.json.zip)

**Comment by rkiesler1** [comment]
The trained eye recognizes the elements of the V3 API, though we should provide some kind of a schema reference to validate the template.

**Comment by jpollock** [comment]
Agreed. I was trying not to build that out quite yet. Not terrible; more important is preventing drift from core API.

**Comment by galois17** [comment]
@jpollock, I saw 403s now and then for exceeding the QPS limit so I added a small tweak to your PR. 

I can confirm things are working great except for a few things I can't check. I'm getting errors with iodocs=true. I'm seeing:

Error: Unable to create the iodocs {"errorCode":404,"errorMessage":"Not Found"}

I'm no Mashery expert. Does this have something to do with my test mashery account? 

Thanks..
